ui/src/app/route-reuse.service.ts

```diff
@@ -0,0 +1,52 @@
+import {Injectable} from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  DetachedRouteHandle,
+  RouteReuseStrategy,
+  Routes
+} from '@angular/router';
+
+// Configure reuse of particular routes so components do not have to re-load
+@Injectable()
+export class RouteReuse implements RouteReuseStrategy {
+
+  public static cached: { [key: string]: any } = {};
+
+  public static routeKey(route: ActivatedRouteSnapshot): string {
+    // We currently only cache the jobs-list route so we can store them based
+    // on just the query parameter. If we add job-detail caching we should
+    // make this key more specific.
+    return route.queryParams.q
+  }
+
+  shouldDetach(route: ActivatedRouteSnapshot): boolean {
+    // Currently we only cache the job list page, we may want to also do
+    // details pages in the future.
+    return !!route.routeConfig && route.routeConfig.path == "jobs";
+  }
+
+  store(route: ActivatedRouteSnapshot, handle: DetachedRouteHandle): void {
+    let storedRoute = {
+        snapshot: route,
+        handle: handle
+    };
+    RouteReuse.cached[RouteReuse.routeKey(route)] = storedRoute;
+  }
+
+  shouldAttach(route: ActivatedRouteSnapshot): boolean {
+    return !!route.routeConfig
+      && route.routeConfig.path == "jobs"
+      && !!RouteReuse.cached[RouteReuse.routeKey(route)];
+  }
+
+  retrieve(route: ActivatedRouteSnapshot): DetachedRouteHandle {
+    if (!route.routeConfig || !RouteReuse.cached[RouteReuse.routeKey(route)]) {
+      return null;
+    }
+    return RouteReuse.cached[RouteReuse.routeKey(route)].handle;
+  }
+
+  shouldReuseRoute(future: ActivatedRouteSnapshot, curr: ActivatedRouteSnapshot): boolean {
```

Comment:
The docs aren't very informative on this one, but I'm slightly surprised by the implementation of `return false;`. Is this just meant to answer whether we can reuse a route for a particular transition (curr->future)?

I'm thinking perhaps you should copy the default impl here, does that cause issues?:
https://github.com/angular/angular/blob/9141424ac69f8ff9702485e7f00e58fa503d8c42/packages/router/src/route_reuse_strategy.ts#L70

Reply:
Yeah I wasn't 100% sure what this is actually used for but I think its if the next component being navigated to is the same as the current one it allows it to be reused instead of recreating. I think this is kind of separate from the caching we're doing. But you're right I can use the default implementation for this. Although it doesn't fix the resolver issue unfortunately.